File: src/main/java/org/example/eventhub/dto/event/EventUpdateRequest.java
package org.example.eventhub.dto.event;

import jakarta.validation.constraints.Positive;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import org.example.eventhub.dto.location.LocationUpdateRequest;
import org.example.eventhub.enums.EventStatus;

public record EventUpdateRequest(
        String title, // TODO сделать проверку что не пустая
        String description,
        LocalDateTime dateTime,
        LocationUpdateRequest location,
        @Positive(message = "capacity должна быть положительна") Integer capacity,
        BigDecimal price,
        EventStatus eventStatus) {}

File: src/main/java/org/example/eventhub/dto/event/EventUpdateRequest.kt
package org.example.eventhub.dto.event

import jakarta.validation.constraints.Positive
import org.example.eventhub.dto.location.LocationUpdateRequest
import org.example.eventhub.enums.EventStatus
import java.math.BigDecimal
import java.time.LocalDateTime

data class EventUpdateRequest(
    val title: String?,  // TODO сделать проверку что не пустая
    val description: String?,
    val dateTime: LocalDateTime?,
    val location: LocationUpdateRequest?,
    val capacity: @Positive(message = "capacity должна быть положительна") Int?,
    val price: BigDecimal?,
    val eventStatus: EventStatus?
)
